analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PreferGuidEmpty.Fixed.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+
+class Compliant
+{
+    Guid? nullableField; // Compliant
+    readonly Guid? readonlyNullableField; // Compliant
+    Guid Property { get; set; } // Compliant
+
+    void Empty()
+    {
+        var empty = Guid.Empty; // Compliant
+    }
+
+    void NotEmpty()
+    {
+        var rnd = Guid.NewGuid();// Compliant
+        var bytes = new Guid(new byte[0]); // Compliant
+        var str = new Guid("FA97FFE7-532C-4015-8698-49D8CE4126F4"); // Compliant
+    }
+
+    void NullableDefault()
+    {
+        Guid? nullable = default; // Compliant, not equivalent to Guid.Empty.
+        var nullableOfT = default(Guid?); // Compliant
+        var guidAsParameter = new NullableGuidClass(default); // Compliant
+        guidAsParameter.Method(default); // Compliant
+    }
+
+    void NotInitialized(string str)
+    {
+        Guid parsed; // Compliant
+        Guid.TryParse(str, out parsed);
+    }
+
+    void OptionalParameter(Guid id = default) { } // Compliant, default has to be a run-time constant
+
+    void NotGuid()
+    {
+        int integer = default;
+        var date = default(DateTime);
+    }
+}
+
+class NonCompliant
+{
+    Guid field; // FN
+    readonly Guid readonlyField; // FN
```

Comment:
I would not mark these as FNs. There's a high chance that they are initialized somewhere.